File: hw/rojas/u1/hw16IncomeTax/incomeTaxesCalculator.py
#author Josue Rojas
#UseofLibrary
from incomeTaxesCalculatorlib import IncomeTaxEC

def run():
    print("----- Welcome to Income Tax Calculation -----")
    calc = IncomeTaxEC()

    while True:
        try:
            monthly = float(input("Enter your monthly salary:\n").strip())
            spent = float(input("Enter your annual personal expenses:\n").strip())
            deps = int(input("Enter your number of family dependents:\n").strip())

            result = calc.compute(monthly_salary=monthly, annual_spent=spent, dependents=deps)

            print(f"\nYour Tax to pay this year is: {round(result.annual_tax, 1)}")

        except Exception as e:
            print("Input error:", e)

        again = input("Would you like to calculate again? (s/n): ").strip().lower()
        if again != "s":
            print("Goodbye!")
            break

if __name__ == "__main__":
    run()

File: hw/rojas/u1/hw16IncomeTax/incomeTaxescalculator.js
//Author: Josue Rojas
// useOfLibrary.js

const { computeIncomeTaxEC } = require("./incomeTaxesCalculatorlib");
const readline = require("readline");

const rl = readline.createInterface({ input: process.stdin, output: process.stdout });
const ask = q => new Promise(res => rl.question(q, res));

async function main() {
  console.log("----- Welcome to Income Tax Calculation -----");

  while (true) {
    try {
      const monthly = parseFloat((await ask("Enter your monthly salary:\n")).trim());
      const spent   = parseFloat((await ask("Enter your annual personal expenses:\n")).trim());
      const deps    = parseInt((await ask("Enter your number of family dependents:\n")).trim(), 10);

      const r = computeIncomeTaxEC(monthly, spent, isNaN(deps) ? 0 : deps);
      console.log(`\nYour Tax to pay this year is: ${r.annualTax.toFixed(1)}\n`);
      // console.log(`Monthly withholding: ${r.monthlyWithholding.toFixed(2)}`);
    } catch (e) {
      console.log("Input error:", e.message);
    }

    const again = (await ask("Would you like to calculate again? (s/n): ")).trim().toLowerCase();
    if (again !== "s") { console.log("Goodbye!"); break; }
  }
  rl.close();
}

main();
